bundler/lib/bundler/compact_index_client/cache.rb

```diff
@@ -54,15 +54,14 @@ def versions_etag_path
         directory.join("versions.etag")
       end
 
-      def checksums
-        checksums = {}
+      def info_checksum_for(name)
+        @version_data ||= versions_path.read.freeze
```

Comment:
I think this is fine, but I want to note that the Cache instance is not retaining data. This is consistent with how we use the class now, but since cache is not responsible for invalidating or refetching `versions`, it can't bust its cache if the file is updated after being read. I think this is actually a design flaw with how the client is implemented but I wanted to keep this PR simple.

Reply:
since the client is what currently holds state, I think it makes sense to move this caching up into the compact index client, and keep the Cache responsible only for reading from disk

Reply:
Parser holds state now after the refactor, which is where state was held when we parsed the entire thing into a big hash.